ddtrace/contrib/botocore/patch.py

```diff
@@ -328,16 +347,37 @@ def inject_trace_to_kinesis_stream_data(record, span):
         record["Data"] = data_json
 
 
-def inject_trace_to_kinesis_stream(params, span):
-    # type: (List[Any], Span) -> None
+def record_data_streams_path_for_kinesis_stream(pin, params, results):
+    stream_arn = params.get("StreamARN")
+
+    if not stream_arn:
+        log.debug("Unable to determine StreamARN for request with params: ", params)
+        return
+
+    processor = pin.tracer.data_streams_processor
+    for record in results.get("Records", []):
+        time_estimate = record.get("ApproximateArrivalTimestamp", datetime.now()).timestamp()
+        processor.new_pathway(now_sec=time_estimate).set_checkpoint(
```

Comment:
I don't think we need to create a separate pathway for every message. Since they all came from the same destination it would be better to create a single pathway object and set checkpoints for every message. This means we'd need to pass the `time_estimate` to `set_checkpoint` (The pathway object has logic to deal with repeating checkpoints correctly - it will disambiguate baed on the direction tag)